File: problems/88/solution.go
package main
func merge(nums1 []int, m int, nums2 []int, n int)  {
    pos := len(nums1) - 1
    m--
    n--
    
    for m >= 0 && n >= 0 {
        if nums1[m] > nums2[n] {
            nums1[pos] = nums1[m]
            m--
        } else {
            nums1[pos] = nums2[n]
            n--
        }
        
        pos--
    }
    
    for m >= 0 {
        nums1[pos] = nums1[m]
        m--
        pos--
    }
    
    for n >= 0 {
        nums1[pos] = nums2[n]
        n--
        pos--
    }
}

File: problems/88/solution.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let pos = nums1.length - 1;
    m--;
    n--
    
    while (~m && ~n) {
        nums1[pos--] = nums1[m] > nums2[n] ? nums1[m--] : nums2[n--];
    }
    
    while (~m) {
        nums1[pos--] = nums1[m--];
    }
    
    while (~n) {
        nums1[pos--] = nums2[n--]
    }
};